src/shiv/cli.py

```diff
@@ -150,27 +167,29 @@ def main(
             if supplied_arg in disallowed:
                 sys.exit(DISALLOWED_PIP_ARGS.format(arg=supplied_arg, reason=DISALLOWED_ARGS[disallowed]))
 
+    sources: List[Path] = []
     with TemporaryDirectory() as tmp_site_packages:
 
         # If both site_packages and pip_args are present, we need to copy the site_packages
         # dir into our staging area (tmp_site_packages) as pip may modify the contents.
         if site_packages:
             if pip_args:
-                _copytree(Path(site_packages), Path(tmp_site_packages))
+                for sp in site_packages:
+                    _copytree(Path(sp), Path(tmp_site_packages))
             else:
-                tmp_site_packages = site_packages
+                sources.extend(map(lambda p: Path(p).expanduser(), site_packages))
```

Comment:
In general, a list comprehension is preferred over lambda:

sources.extend([Path(p).expanduser() for p in site_packages])